### PR TITLE
Fix back end layout problems in various browsers

### LINKED ISSUE #1721: Some back end widgets break layout in Firefox 76
**Affected version(s)**

Contao 4.9.2

**Description**

Start point of website

Sub palette "DNS settings"

language and fallback language are not in the correct column.
grafik

### PR DESCRIPTION
**Pull request by fritzmg**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #1721
| Docs PR or issue | N/A

This fixes the layout problems mentioned in #1721 in various browsers, zoom levels and device pixel density settings. Tested on the recent versions of Firefox, Chrome and Edge under Windows.

### CODE REVIEW

**Review by ausi — COMMENTED** [review]
Did you forget to update the non-minified version *src/basic.css*?

**Comment by fritzmg** [comment]
Oh, there they are 😄. Updated.

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/Resources/contao/themes/flexible/src/basic.css [hunk 1]
```diff
@@ -227,6 +227,7 @@ fieldset.checkbox_container,fieldset.radio_container {
 	width:19%;
 }
 .tl_text,.tl_text_2,.tl_text_3,.tl_text_4,.tl_textarea,.tl_text_unit,.tl_text_trbl,.tl_text_interval {
+	height:29px;
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/themes/flexible/src/basic.css]: I'm afraid this is wrong. The fields should be 30px high.

File: core-bundle/src/Resources/contao/themes/flexible/src/basic.css [hunk 2]
```diff
@@ -299,6 +300,7 @@ select[multiple] {
 	width:50%;
 }
 .tl_select,.tl_mselect,.tl_select_column,.tl_select_unit,.tl_select_interval {
+	height:29px;
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/themes/flexible/src/basic.css]: Again, this should be `height: 30px`.

File: core-bundle/src/Resources/contao/themes/flexible/src/basic.css [hunk 3]
```diff
@@ -539,7 +541,7 @@ a.tl_submit {
 	margin:16px 15px 8px;
 }
 .cbx {
-	min-height:56px;
+	min-height:57px;
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/themes/flexible/src/basic.css]: This increased the overall height of these fields to 81px, although the text and select widgets are only 80px high. Tested on Mac OS with Safari, Chrome and Firefox.

**Review by leofeyer — CHANGES_REQUESTED** [review]

**Comment by fritzmg** [comment]
> I'm afraid this is wrong. The fields should be 30px high.

They should (10px padding + 18px line-height + 2px border = 30px), but unfortunately, it's not the case across all browsers, operating systems, zoom levels and device pixel densities … choosing `30px` breaks the layout again in some of these combinations.

**Comment by leofeyer** [comment]
But choosing `29px` decreases the height of the fields and that is not acceptable, either.

**Comment by fritzmg** [comment]
Then we need another approach I think. Switch to `display: flex`? ;)

**Comment by leofeyer** [comment]
Sure, if you want to spend some time working on this. 😄 I'd love using CSS grid and/or flex for the back end. Then we could also have more than 2 columns.

**Comment by m-vo** [comment]
Imo it should be `grid` then, not `flex`. But I'd :heart: to see this.

**Comment by fritzmg** [comment]
My experience with CSS Grid is very limited and I still don't understand most of it, so someone else would have to do this 😁 

Regarding the original issue: what if we simply defined the `min-height` of each widget to be `81px`? In my opinion we will never be able to make each individual component within a widget to be exactly a specific height, so that is sums up to 80px or less in all circumstances. Sure, it would be a dirty fix - but it's a dirty fix for an old layout - and that seems fine to me.

**Comment by leofeyer** [comment]
> what if we simply defined the `min-height` of each widget to be `81px`?

I guess that is worth a try.

**Comment by m-vo** [comment]
Just for the record: here is a short POC how css grid could help in this situation: https://jsfiddle.net/k8fshxev - it would break if someone created own things (like a "w33" element) of course.

**Comment by leofeyer** [comment]
@fritzmg I suppose this PR needs to be adjusted according to #1823, doesn't it?

**Comment by fritzmg** [comment]
Yes, will do.

**Comment by leofeyer** [comment]
Now we will also have to incorporate the changes from #1875. 🙈

**Comment by fritzmg** [comment]
The most recent commit incorporates the changes for 4.9 in 4.4.

The change to `.tl_box h3,.tl_tbox h3,.tl_xpl h3` and the accompanying increase in margin from `2px` to `3px` for the text, select and TinyMCE inputs is something that was not when switching from 4.4 to 4.9. I had to incorporate it here for 4.4 too, otherwise FireFox would have again rendered some elements at different heights due to subpixels.

File: core-bundle/src/Resources/contao/themes/flexible/src/basic.css [hunk 4]
```diff
@@ -536,10 +540,15 @@ a.tl_submit {
 	width:100%;
 }
 .m12 {
-	margin:16px 15px 8px;
+	margin:0 15px;
+	padding:17px 0;
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/themes/flexible/src/basic.css:544]: I thought you wanted to revert the changes to the `m12` class so we do not have to add a `.cbx.m12` format definition?

**Review by leofeyer — COMMENTED** [review]
Compared to the `4.9` branch, I am missing this change:



Did you leave it out on purpose?

File: core-bundle/src/Resources/contao/themes/flexible/src/basic.css (hunk 4 above)
Comment (fritzmg) [reply core-bundle/src/Resources/contao/themes/flexible/src/basic.css:544]: Yes, but I had to keep it, otherwise layout will break again.

**Comment by fritzmg** [comment]
> Did you leave it out on purpose?

No, I missed that one. I'll add it (and check it).

File: core-bundle/src/Resources/contao/themes/flexible/src/basic.css (hunk 4 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/themes/flexible/src/basic.css:544]: I see. 👍

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @fritzmg.

**Pull request merged**